docs/Contributing/Testing-and-local-development.md

```diff
@@ -448,3 +449,123 @@ MinIO also offers a web interface at http://localhost:9001. Credentials are `min
+
+1. A private key.
+1. A certificate.
+1. An encrypted token generated by Apple.
+
+### Private key + certificate
+
+You can generate the private key and the certificate using `fleetctl`:
+
+```
+fleetctl generate mdm-apple-bm
+```
+
+This will output two files `fleet-apple-mdm-bm-public-key.crt` and `fleet-apple-mdm-bm-private.key`, save them in a safe place.
+
+### Encrypted token
+
+Ask @GuillaumeRoss to create an account for you in [ABM](https://github.com/fleetdm/fleet/blob/main/tools/mdm/apple/glossary-and-protocols.md#abm-apple-business-manager) 
+
+Once you have access:
+
+1. Go to https://business.apple.com/#/main/preferences/myprofile
+1. Click on "+ Add" to create a new MDM server.
+1. Use a name that allows you to identify the server.
+1. Under "Upload Public Key," upload the `fleet-apple-mdm-bm-public-key.crt` you generated before.
+1. Click "Save."
+1. Click on the "Download Token" button at the top and confirm the download in the modal.
+1. Save the token in a safe place.
+
+### APNs setup
+
+The server also needs a private key + certificate to identify with Apple's [APNs](https://github.com/fleetdm/fleet/blob/main/tools/mdm/apple/glossary-and-protocols.md#apns-apple-push-notification-service) servers.
+
+Users will be able to generate this using `fleetctl generate mdm-apple` in the near future, but for now, ask in [#g-mdm](https://fleetdm.slack.com/archives/C03C41L5YEL) how to get them.
+
+### SCEP setup
+
+Another key pair is used for [SCEP](https://github.com/fleetdm/fleet/blob/main/tools/mdm/apple/glossary-and-protocols.md#scep-simple-certificate-enrollment-protocol), in the near future, this will also be generated by `fleetctl generate mdm-apple`, but for now please use this:
+
+```
+openssl genrsa -out fleet-mdm-apple-scep.key 4096
+
+openssl req -x509 -new -nodes -key fleet-mdm-apple-scep.key -sha256 -days 1826 -out fleet-mdm-apple-scep.crt -subj '/CN=Fleet Root CA/C=US/O=Fleet DM.'
+```
+
+### Running the server
+
+Try to store all the certificates and tokens you generated in the earlier steps together in a safe place outside of the repo, then start the server with:
+
+```
+FLEET_MDM_APPLE_ENABLE=1 \
+FLEET_MDM_APPLE_SCEP_CHALLENGE=scepchallenge \
+FLEET_MDM_APPLE_SCEP_CERT=/path/to/fleet-mdm-apple-scep.crt \
+FLEET_MDM_APPLE_SCEP_KEY=/path/to/fleet-mdm-apple-scep.key \
+FLEET_MDM_APPLE_BM_SERVER_TOKEN=/path/to/dep_encrypted_token.p7m \
+FLEET_MDM_APPLE_BM_CERT=/path/to/fleet-apple-mdm-bm-public-key.crt \
+FLEET_MDM_APPLE_BM_KEY=/path/to/fleet-apple-mdm-bm-private.key \
+FLEET_MDM_APPLE_APNS_CERT=/path/to/mdmcert.download.push.pem \
+FLEET_MDM_APPLE_APNS_KEY=/path/to/mdmcert.download.push.key \
+ ./build/fleet serve --dev --dev_license --logging_debug
+```
+
+The server needs to run behind TLS with a valid certificate, so in a separate terminal window/tab, create a local tunnel to your server using `ngrok` (`brew install ngrok/ngrok/ngrok` if you don't have it.)
```

Comment:
Do you know what specific features require this approach? For the basic usage of e.g. making requests to `GET /mdm/apple` and `GET /mdm/apple_bm`, this should not be required (i.e. it's ok to run with a localhost insecure/self-signed cert).

Reply:
good point, I will clarify. This is needed if you need to enroll a VM (and thus any feature that requires enrollment)